src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -2344,6 +2344,12 @@ private BoundExpression BindExplicitNullableCastFromNonNullable(ExpressionSyntax
             {
                 bag.Free();
             }
+
+            ErrorCode getErrorCode(DiagnosticBag diagnostic)
+            {
+                int code = diagnostic.AsEnumerable().LastOrDefault().Code;
```

Comment:
It seems like this local function is called with the expectation of a non-empty bag, so if we're going to use this approach then let's use .Last()